src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs

```diff
@@ -223,11 +189,20 @@ public class AdminController : Controller
                 AllowHybridFlow = model.AllowHybridFlow
             };
 
-            await _applicationManager.CreateAsync(openIdApp);
+            if (model.Type == ClientType.Confidential)
+            {
+                await _applicationManager.CreateAsync(application, model.ClientSecret, HttpContext.RequestAborted);
```

Comment:
In the commit I'm working on based on what we comment here I've added a dirty check that works. 
```
public Task SetClientTypeAsync(OpenIdApplication application, string type, CancellationToken cancellationToken)
        {
            if (application == null)
            {
                throw new ArgumentNullException(nameof(application));
            }

            if (type == null || type.Length<1)
            {
                throw new ArgumentNullException(nameof(type));
            }

            type= type[0].ToString().ToUpper() + type.Substring(1);
            application.Type = (ClientType) Enum.Parse(typeof(ClientType), type);

            return Task.CompletedTask;
        }
```